### PR TITLE
Fix "Undefined array key _ in Otl.php:1542"

### LINKED ISSUE #1963: Undefined array key _ in Otl.php:1542
### Guidelines

- [X] I understand that [if I fail to adhere to contribution guidelines and/or fail to provide all required details, this issue may be closed without review](https://github.com/mpdf/mpdf/blob/development/.github/CONTRIBUTING.md).

### Description of the bug

When trying to add certain characters to PDF content, such as `न्`, I get a PHP warnings about a non-existent offset.
```
Warning: Undefined array key 2 in /application/vendor/mpdf/mpdf/src/Otl.php on line 1542
Warning: Trying to access array offset on value of type null in /application/vendor/mpdf/mpdf/src/Otl.php on line 1542
Warning: Undefined array key 2 in /application/vendor/mpdf/mpdf/src/Otl.php on line 1543
Warning: Trying to access array offset on value of type null in /application/vendor/mpdf/mpdf/src/Otl.php on line 1543
```

The file is still generated and visually has no problems. So I suggest adding a check for the existence of an offset in order to avoid warnings

### mPDF version

8.1.2 (any)

### PHP Version and environment (server type, cli provider etc., enclosing libraries and their respective versions)

PHP 8.2

### Reproducible PHP+CSS+HTML snippet suffering by the error

```php
$mpdf = new Mpdf([
        'mode' => '-aCJK',
        'autoScriptToLang' => true,
        'autoLangToFont' => true,
        'default_font' => 'dejavusans',
]);

$mpdf->WriteHTML('<p>न्</p>');
$mpdf->Output('', Destination::STRING_RETURN);
```

### PR DESCRIPTION
**Pull request by sonnymilton**
Checking the existence of the offset

Closes #1963

### CODE REVIEW

**Comment by sonnymilton** [comment]
Hello @finwe,

I have added the test case as you requested in the test/Issues directory. This test checks for the absence of warnings when creating a PDF, as related to my changes.

Please let me know if anything else is needed or if you have any comments/suggestions on my PR.

File: src/Otl.php [hunk 1]
```diff
@@ -1535,7 +1535,7 @@ function _applyGSUBrulesIndic($usetags, $scriptTag, $langsys, $is_old_spec)
 										$mask = (1 << (Indic::PSTF));
 										break;
 								}
-								if (!($this->OTLdata[$ptr]['mask'] & $mask)) {
+								if (!($this->OTLdata[$ptr]['mask'] & $mask) || !isset($this->OTLdata[$ptr + 1])) {
```
Comment (finwe) [inline src/Otl.php:1538]: Can you make the condition separate so that the code is more readable?

File: tests/Issues/Issue1963Test.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace Issues;
+
+use Mpdf\BaseMpdfTest;
+use Mpdf\Mpdf;
+use Mpdf\Output\Destination;
+
+class Issue1963Test extends BaseMpdfTest
+{
+	protected function set_up()
```
Comment (finwe) [inline tests/Issues/Issue1963Test.php:12]: This method is kind of unnecessary, make a local variable directly in the test method.

File: tests/Issues/Issue1963Test.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+			'autoScriptToLang' => true,
+			'autoLangToFont' => true,
+			'default_font' => 'dejavusans',
+		]);
+	}
+
+	public function testNoWarning()
+	{
+		$this->mpdf->WriteHTML('<p>न्</p>');
+		$output = $this->mpdf->Output('my.pdf', Destination::STRING_RETURN);
```
Comment (finwe) [inline tests/Issues/Issue1963Test.php:25]: Maybe use the shorthand [OutputBinaryData()](https://mpdf.github.io/reference/mpdf-functions/outputbinarydata.html) directly? the filename is unnecessary here.

File: tests/Issues/Issue1963Test.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+
```
Comment (finwe) [inline tests/Issues/Issue1963Test.php:3]: Only a single empty line, please (should have been caught in code style check)

**Review by finwe — CHANGES_REQUESTED** [review]
Looking fine apart from a few minor enhancements. Please resolve these, squash the commits and I'll be happy to merge.

**Comment by sonnymilton** [comment]
@finwe,

I have made the requested changes to the PR. 
Please review the latest updates and let me know if anything else is required.

**Comment by sonnymilton** [comment]
Hello @finwe,

I am glad that my PR has been merged. I believe that many library users could benefit from this change.

I would like to ask you to publish a new version with these changes so that I and other users can take advantage of these improvements in our projects.

**Pull request merged**